pages/api/id/[form]/submission/confirm.ts

```diff
@@ -0,0 +1,210 @@
+      `User does not have an associated email address: ${JSON.stringify(session.user)} `
+    );
+
+  const formId = req.query.form;
+
+  if (Array.isArray(formId) || !formId) return res.status(400).json({ error: "Bad request" });
+
+  const confirmationCodes = req.body as string[];
+
+  try {
+    const dynamoDbClient = connectToDynamo();
+
+    const submissionsFromConfirmationCodes = await getSubmissionsFromConfirmationCodes(
+      formId,
+      confirmationCodes,
+      dynamoDbClient
+    );
+
+    if (submissionsFromConfirmationCodes.submissionsToConfirm.length > 0) {
+      await confirm(formId, submissionsFromConfirmationCodes.submissionsToConfirm, dynamoDbClient);
+    }
+
+    logMessage.info(
+      `user:${userEmail} confirmed form submissions [${submissionsFromConfirmationCodes.submissionsToConfirm.map(
+        (submission) => submission.name
+      )}] for form ID:${formId} at:${Date.now()}`
+    );
+
+    return res.status(200).json({
+      ...(submissionsFromConfirmationCodes.submissionsToConfirm.length > 0 && {
+        confirmedSubmissions: submissionsFromConfirmationCodes.submissionsToConfirm.map(
+          (submission) => submission.confirmationCode
+        ),
+      }),
+      ...(submissionsFromConfirmationCodes.confirmationCodesAlreadyUsed.length > 0 && {
+        confirmationCodesAlreadyUsed: submissionsFromConfirmationCodes.confirmationCodesAlreadyUsed,
+      }),
+      ...(submissionsFromConfirmationCodes.confirmationCodesNotFound.length > 0 && {
+        invalidConfirmationCodes: submissionsFromConfirmationCodes.confirmationCodesNotFound,
+      }),
+    });
+  } catch (error) {
+    logMessage.error(error as Error);
+    return res.status(500).json({ error: "Error on server side" });
+  }
+};
+
+async function getSubmissionsFromConfirmationCodes(
+  formId: string,
+  confirmationCodes: string[],
+  dynamoDbClient: DynamoDBDocumentClient
+): Promise<{
+  submissionsToConfirm: { name: string; confirmationCode: string }[];
+  confirmationCodesAlreadyUsed: string[];
+  confirmationCodesNotFound: string[];
+}> {
+  const accumulatedSubmissions: {
+    [confCode: string]: { name: string; removalDate?: number };
+  } = {};
+
+  let requestedKeys = confirmationCodes.map((code) => {
+    return { FormID: formId, NAME_OR_CONF: `CONF#${code}` };
+  });
+
+  while (requestedKeys.length > 0) {
+    const request = new BatchGetCommand({
+      RequestItems: {
+        Vault: {
+          Keys: requestedKeys,
+          ProjectionExpression: "NAME_OR_CONF,#name,RemovalDate",
+          ExpressionAttributeNames: {
+            "#name": "Name",
+          },
+        },
+      },
+    });
+
+    // eslint-disable-next-line no-await-in-loop
+    const response = await dynamoDbClient.send(request);
+
+    if (response.Responses && response.Responses["Vault"]) {
+      response.Responses["Vault"].forEach((record) => {
+        const confCode = (record["NAME_OR_CONF"] as string).substring(5); // Removing 'CONF#'
```

Comment:
@bryan-robitaille Not sure if it is worth changing but if we add the `ConfirmationCode` attribute also in entries of type `CONF#` we would not have to do that `substring`. I guess we would also be consistent with what we said yesterday about having key attributes (to deal with requests) and "read" attributes for everything else.

Reply:
Sure, I don't see an issue with adding the `ConfirmationCode` as an attribute to the `CONF#` entry.  I can make the change in the other draft PR.

Reply:
Alright I will make the necessary change here.